internal/plugin/github.go

```diff
@@ -5,22 +5,45 @@ import (
 	"io"
 	"net/http"
 	"net/url"
+	"regexp"
+	"strings"
 
+	"github.com/samber/lo"
 	"go.jetpack.io/devbox/internal/boxcli/usererr"
 	"go.jetpack.io/devbox/internal/cachehash"
 	"go.jetpack.io/devbox/nix/flake"
 )
 
 type githubPlugin struct {
-	ref flake.Ref
+	ref  flake.Ref
+	name string
+}
+
+// Github only allows alphanumeric, hyphen, underscore, and period in repo names.
+// but we clean up just in case.
```

Comment:
why is the "but we clean up just in case" comment needed? The regex checks for the all the cases that Github allows for, which is good. Not understanding the "but..." as if we are doing some further restrictions?

Reply:
I just meant that we should never encounter weird stuff (because it would not be a valid github repo) but we sanitize it anyway just in case. (for example we really don't want a slash in there)